trino-aws-proxy-spi/src/main/java/io/trino/aws/proxy/spi/plugin/TrinoAwsProxyServerBinding.java

```diff
@@ -56,6 +58,8 @@ static <T extends Identity> void bindIdentityType(Binder binder, Class<T> type)
             log.info("Using %s identity type", type.getSimpleName());
             return type;
         });
+        newSetBinder(binder, com.fasterxml.jackson.databind.Module.class).addBinding()
```

Comment:
This looks reasonable, but we are now exposing Jackson on the SPI. I don't have a strong opinion but would like to know what @Randgalt and @mosiac1 think about this.

We could instead, inside `TrinoAwsProxyServerModule` add:

```java
    public static class JsonIdentityProvider implements Provider<com.fasterxml.jackson.databind.Module>
    {
        private final Class<? extends Identity> identityType;

        @Inject
        public JsonIdentityProvider(Class<? extends Identity> identityType)
        {
            this.identityType = identityType;
        }

        @Override
        public com.fasterxml.jackson.databind.Module get()
        {
            return new SimpleModule().addAbstractTypeMapping(Identity.class, identityType);
        }
    }
```

And bind it on creation of the server. That way we don't expose any json binding to the SPI

Reply:
+1 for doing it in the `TrinoAwsProxyServerModule` rather than in the plugin binding, this seems to me like an implementation detail